networkdb/delegate.go

```diff
@@ -71,19 +72,33 @@ func (nDB *NetworkDB) handleNodeEvent(nEvent *NodeEvent) bool {
 	nDB.purgeSameNode(n)
 	n.ltime = nEvent.LTime
 
+	var op opType
+	broadcast := false
+
 	switch nEvent.Type {
 	case NodeEventTypeJoin:
 		nDB.Lock()
 		nDB.nodes[n.Name] = n
 		nDB.Unlock()
-		return true
+		op = opCreate
+		broadcast = true
 	case NodeEventTypeLeave:
 		nDB.Lock()
 		nDB.leftNodes[n.Name] = n
 		nDB.Unlock()
-		return true
+		op = opDelete
+		broadcast = true
 	}
 
+	if broadcast {
+		value, err := json.Marshal(&NodeAddr{n.Addr})
```

Comment:
Why ? This is a local event between the networkdb & the controller. It can be anything that generates a []byte. Its enough if networkdb & controller agrees upon.